{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "#from scipy.stats import norm\n",
    "\n",
    "import pybedtools\n",
    "import re\n",
    "#import matplotlib as plt\n",
    "import pyBigWig as pbw\n",
    "import time\n",
    "import urllib.request\n",
    "import matplotlib.pyplot as plt\n",
    "import json\n",
    "import requests\n",
    "from pyliftover import LiftOver"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                       study_id           pmid    pub_date pub_journal  \\\n",
      "773           FINNGEN_R5_E4_DM2                 2021-05-11               \n",
      "774       FINNGEN_R5_E4_DM2COMA                 2021-05-11               \n",
      "776    FINNGEN_R5_E4_DM2NASCOMP                 2021-05-11               \n",
      "778     FINNGEN_R5_E4_DM2NOCOMP                 2021-05-11               \n",
      "779       FINNGEN_R5_E4_DM2OPTH                 2021-05-11               \n",
      "780     FINNGEN_R5_E4_DM2PERIPH                 2021-05-11               \n",
      "782    FINNGEN_R5_E4_DM2_STRICT                 2021-05-11               \n",
      "2793             FINNGEN_R5_T2D                 2021-05-11               \n",
      "2794     FINNGEN_R5_T2D_INCLAVO                 2021-05-11               \n",
      "2795        FINNGEN_R5_T2D_WIDE                 2021-05-11               \n",
      "11388                GCST005047  PMID:22885922  2012-09-01   Nat Genet   \n",
      "11719                GCST005413  PMID:29358691  2018-01-22  Nat Commun   \n",
      "12678                GCST005898  PMID:29358691  2018-01-22  Nat Commun   \n",
      "15762                GCST006867  PMID:30054458  2018-07-27  Nat Commun   \n",
      "16734                GCST007517  PMID:29632382  2018-04-09   Nat Genet   \n",
      "16735                GCST007518  PMID:29632382  2018-04-09   Nat Genet   \n",
      "30674         NEALE2_20002_1223                 2018-08-01               \n",
      "32677               SAIGE_250_2                 2018-10-24               \n",
      "32680              SAIGE_250_23                 2018-10-24               \n",
      "\n",
      "                                               pub_title      pub_author  \\\n",
      "773                                                           FINNGEN_R5   \n",
      "774                                                           FINNGEN_R5   \n",
      "776                                                           FINNGEN_R5   \n",
      "778                                                           FINNGEN_R5   \n",
      "779                                                           FINNGEN_R5   \n",
      "780                                                           FINNGEN_R5   \n",
      "782                                                           FINNGEN_R5   \n",
      "2793                                                          FINNGEN_R5   \n",
      "2794                                                          FINNGEN_R5   \n",
      "2795                                                          FINNGEN_R5   \n",
      "11388  Large-scale association analysis provides insi...       Morris AP   \n",
      "11719  Re-analysis of public genetic data reveals a r...  Bonas-Guarch S   \n",
      "12678  Re-analysis of public genetic data reveals a r...  Bonas-Guarch S   \n",
      "15762  Genome-wide association analyses identify 143 ...           Xue A   \n",
      "16734  Refining the accuracy of validated target iden...       Mahajan A   \n",
      "16735  Refining the accuracy of validated target iden...       Mahajan A   \n",
      "30674                                                       UKB Neale v2   \n",
      "32677                                                          UKB SAIGE   \n",
      "32680                                                          UKB SAIGE   \n",
      "\n",
      "                                          trait_reported  \\\n",
      "773                                      Type 2 diabetes   \n",
      "774                            Type 2 diabetes with coma   \n",
      "776    Type 2 diabetes with other specified/multiple/...   \n",
      "778                Type 2 diabetes without complications   \n",
      "779        Type 2 diabetes with ophthalmic complications   \n",
      "780    Type 2 diabetes with peripheral circulatory co...   \n",
      "782              \"Type 2 diabetes, strict (exclude DM1)\"   \n",
      "2793             \"Type 2 diabetes, definitions combined\"   \n",
      "2794   \"Type 2 diabetes, definitions combined, includ...   \n",
      "2795                  \"Type 2 diabetes, wide definition\"   \n",
      "11388                                    Type 2 diabetes   \n",
      "11719                                    Type 2 diabetes   \n",
      "12678                                    Type 2 diabetes   \n",
      "15762                                    Type 2 diabetes   \n",
      "16734                                    Type 2 diabetes   \n",
      "16735                 Type 2 diabetes (adjusted for BMI)   \n",
      "30674  Type 2 diabetes | non-cancer illness code, sel...   \n",
      "32677                                    Type 2 diabetes   \n",
      "32680     Type 2 diabetes with ophthalmic manifestations   \n",
      "\n",
      "                                              trait_efos  \\\n",
      "773    [EFO_0001360, HP_0001259, EFO_1000897, EFO_000...   \n",
      "774                            [EFO_0001360, HP_0001259]   \n",
      "776                           [EFO_0001360, EFO_0009518]   \n",
      "778                                        [EFO_0001360]   \n",
      "779                           [EFO_0001360, EFO_0009518]   \n",
      "780                           [EFO_0001359, EFO_0003875]   \n",
      "782                                        [EFO_0001360]   \n",
      "2793                                       [EFO_0001360]   \n",
      "2794                                       [EFO_0001360]   \n",
      "2795                                       [EFO_0001360]   \n",
      "11388                                      [EFO_0001360]   \n",
      "11719                                      [EFO_0001360]   \n",
      "12678                                      [EFO_0001360]   \n",
      "15762                                      [EFO_0001360]   \n",
      "16734                                      [EFO_0001360]   \n",
      "16735                                      [EFO_0001360]   \n",
      "30674                                      [EFO_0001360]   \n",
      "32677                                      [EFO_0001360]   \n",
      "32680                                      [EFO_0001360]   \n",
      "\n",
      "                          ancestry_initial  \\\n",
      "773                      [European=215654]   \n",
      "774                      [European=185432]   \n",
      "776                      [European=207318]   \n",
      "778                      [European=197807]   \n",
      "779                      [European=185304]   \n",
      "780                      [European=184234]   \n",
      "782                      [European=212351]   \n",
      "2793                     [European=211766]   \n",
      "2794                     [European=209439]   \n",
      "2795                     [European=202046]   \n",
      "11388                     [European=69033]   \n",
      "11719                     [European=70127]   \n",
      "12678                     [European=20979]   \n",
      "15762  [South Asian=3650, European=655666]   \n",
      "16734                    [European=298957]   \n",
      "16735                    [European=298957]   \n",
      "30674                    [European=361141]   \n",
      "32677                    [European=407701]   \n",
      "32680                    [European=390054]   \n",
      "\n",
      "                                    ancestry_replication  n_initial  \\\n",
      "773                                                   []     215654   \n",
      "774                                                   []     185432   \n",
      "776                                                   []     207318   \n",
      "778                                                   []     197807   \n",
      "779                                                   []     185304   \n",
      "780                                                   []     184234   \n",
      "782                                                   []     212351   \n",
      "2793                                                  []     211766   \n",
      "2794                                                  []     209439   \n",
      "2795                                                  []     202046   \n",
      "11388                 [European=77138, South Asian=3650]      69033   \n",
      "11719                                               None      70127   \n",
      "12678  [European=9253, Hispanic or Latin American=349...      20979   \n",
      "15762                                               None     659316   \n",
      "16734                                               None     298957   \n",
      "16735                                               None     298957   \n",
      "30674                                                 []     361141   \n",
      "32677                                                 []     407701   \n",
      "32680                                                 []     390054   \n",
      "\n",
      "       n_replication  n_cases    trait_category  num_assoc_loci  has_sumstats  \n",
      "773              0.0  32469.0    Cardiovascular              60          True  \n",
      "774              0.0   2247.0  Endocrine system               1          True  \n",
      "776              0.0  24133.0  Endocrine system              52          True  \n",
      "778              0.0  14622.0  Endocrine system              27          True  \n",
      "779              0.0   2119.0  Endocrine system               2          True  \n",
      "780              0.0   1049.0    Cardiovascular               2          True  \n",
      "782              0.0  29166.0  Endocrine system              52          True  \n",
      "2793             0.0  29193.0  Endocrine system              51          True  \n",
      "2794             0.0  28717.0  Endocrine system              57          True  \n",
      "2795             0.0  17268.0  Endocrine system              30          True  \n",
      "11388        80788.0  12171.0  Endocrine system              29          True  \n",
      "11719            NaN  12931.0  Endocrine system              16          True  \n",
      "12678       187275.0   5277.0  Endocrine system               1          True  \n",
      "15762            NaN  62892.0  Endocrine system             131          True  \n",
      "16734            NaN  48286.0  Endocrine system              46          True  \n",
      "16735            NaN  48286.0  Endocrine system              56          True  \n",
      "30674            0.0   2292.0  Endocrine system               3          True  \n",
      "32677            0.0  18945.0  Endocrine system              49          True  \n",
      "32680            0.0   1298.0  Endocrine system               2          True  \n"
     ]
    }
   ],
   "source": [
    "# Prepare SNP input files for CHEERS:\n",
    "#Study_table=pd.read_json(\"/home/xg1/genetics-v2d-data/tmp/210601/merged_study_table.json\",orient='records', \n",
    "#                         lines=True)\n",
    "\n",
    "Study_table=pd.read_parquet(\"/Users/xg1/Downloads/v2d_210601_studies.parquet\")\n",
    "Study_table_sumstats=Study_table.loc[Study_table['has_sumstats'] == True]\n",
    "\n",
    "Study_table.head()\n",
    "\n",
    "Bowels_study_table=Study_table_sumstats.loc[Study_table_sumstats['trait_reported'].str.contains(\n",
    "        'type 2 diabetes', flags=re.I, regex=True) & (Study_table_sumstats['num_assoc_loci'] > 0)]\n",
    "\n",
    "#Bowels_study_table=Study_table_sumstats.loc[Study_table_sumstats['study_id'].str.contains(\n",
    "#        'GCST005536', flags=re.I, regex=True) & (Study_table_sumstats['num_assoc_loci'] > 1)]\n",
    "print(Bowels_study_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "study_table=pd.read_parquet(\"/home/xg1/genetics-v2d-data/output/210601/studies.parquet\")\n",
    "Bowels_study_table=study_table.loc[study_table['trait_reported'].str.contains(\n",
    "        'rheumatoid arthritis', flags=re.I, regex=True)]\n",
    "print(Bowels_study_table.to_string())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    tag_chrom   tag_pos\n",
      "0        chr1  39570256\n",
      "1        chr1  39371430\n",
      "2        chr1  39382675\n",
      "3        chr1  39402949\n",
      "4        chr1  39375096\n",
      "..        ...       ...\n",
      "495      chr9  22133985\n",
      "496      chr9  22134173\n",
      "497      chr9  22134652\n",
      "498      chr9  22133646\n",
      "499      chr9  22136441\n",
      "\n",
      "[500 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "Cred_SNPs=pd.read_parquet(\"/Users/xg1/Downloads/finemapping.parquet\",\n",
    "                engine='pyarrow')\n",
    "\n",
    "Select_SNPs=Cred_SNPs.loc[Cred_SNPs['study_id'].str.contains(\"GCST005413\")].reset_index()\n",
    "Select_SNPs=Select_SNPs[['tag_chrom', 'tag_pos']]\n",
    "Select_SNPs['tag_chrom']=\"chr\"+Select_SNPs['tag_chrom']\n",
    "print(Select_SNPs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    tag_chrom   tag_pos\n",
      "0        chr1  39570256\n",
      "1        chr1  39371430\n",
      "2        chr1  39382675\n",
      "3        chr1  39402949\n",
      "4        chr1  39375096\n",
      "..        ...       ...\n",
      "495      chr9  22133985\n",
      "496      chr9  22134173\n",
      "497      chr9  22134652\n",
      "498      chr9  22133646\n",
      "499      chr9  22136441\n",
      "\n",
      "[500 rows x 2 columns]\n",
      "        hg38_names hg38_chr  hg38_pos\n",
      "0    chr1_39104584     chr1  39104584\n",
      "1    chr1_38905758     chr1  38905758\n",
      "2    chr1_38917003     chr1  38917003\n",
      "3    chr1_38937277     chr1  38937277\n",
      "4    chr1_38909424     chr1  38909424\n",
      "..             ...      ...       ...\n",
      "495  chr9_22133986     chr9  22133986\n",
      "496  chr9_22134174     chr9  22134174\n",
      "497  chr9_22134653     chr9  22134653\n",
      "498  chr9_22133647     chr9  22133647\n",
      "499  chr9_22136442     chr9  22136442\n",
      "\n",
      "[500 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "Cred_SNPs=pd.read_parquet(\"/Users/xg1/Downloads/finemapping.parquet\",\n",
    "                engine='pyarrow')\n",
    "\n",
    "Select_SNPs=Cred_SNPs.loc[Cred_SNPs['study_id'].str.contains(\"GCST005413\")].reset_index()\n",
    "Select_SNPs=Select_SNPs[['tag_chrom', 'tag_pos']]\n",
    "Select_SNPs['tag_chrom']=\"chr\"+Select_SNPs['tag_chrom']\n",
    "print(Select_SNPs)\n",
    "\n",
    "lo=LiftOver(\"hg19\", \"hg38\")\n",
    "\n",
    "def Call_liftover(input_chr, input_pos, input_LO):\n",
    "    return(input_LO.convert_coordinate(input_chr, input_pos))\n",
    "\n",
    "def Liftover_chr(input_chr, input_pos, input_LO):\n",
    "    lifted_result=input_LO.convert_coordinate(input_chr, input_pos)\n",
    "    if(len(lifted_result) == 0):\n",
    "        return(\"NA\")\n",
    "    else:\n",
    "        return(lifted_result[0][0])\n",
    "\n",
    "def Liftover_pos(input_chr, input_pos, input_LO):\n",
    "    lifted_result=input_LO.convert_coordinate(input_chr, input_pos)\n",
    "    if(len(lifted_result) == 0):\n",
    "        return(\"NA\")\n",
    "    else:\n",
    "        return(lifted_result[0][1])\n",
    "    \n",
    "\n",
    "Select_SNPs['hg38_chr']=Select_SNPs.apply(lambda x: Liftover_chr(input_chr=x['tag_chrom'], input_pos=x['tag_pos'], input_LO=lo), axis=1)\n",
    "Select_SNPs['hg38_pos']=Select_SNPs.apply(lambda x: Liftover_pos(input_chr=x['tag_chrom'], input_pos=x['tag_pos'], input_LO=lo), axis=1)\n",
    "#Select_SNPs=Select_SNPs.loc[-Select_SNPs['hg38_chr'].str.contains(\"NA\")]\n",
    "\n",
    "Select_SNPs['hg38_names']=Select_SNPs.apply(lambda x: x['hg38_chr']+\"_\"+str(x['hg38_pos']), axis=1)\n",
    "Select_SNPs=Select_SNPs[['hg38_names', 'hg38_chr', 'hg38_pos']]\n",
    "\n",
    "\n",
    "#Select_SNPs['name']=Select_SNPs.apply(lambda x: x['tag_chrom']+\"_\"+str(x['tag_pos']), axis=1)\n",
    "#Select_SNPs=Select_SNPs[['name', 'tag_chrom', 'tag_pos']]\n",
    "\n",
    "print(Select_SNPs)\n",
    "Select_SNPs.to_csv(\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/FINNGEN_R5_L12_ATOPIC_hg38.txt\", sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "#Select_SNPs.to_csv(\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/FINNGEN_R5_L12_ATOPIC.txt\", sep=\"\\t\", index=False, header=False)\n",
    "#Select_SNPs.\n",
    "#Traits_all_SNPs_pp['name']=Traits_all_SNPs_pp.apply(lambda x: x['name']+\"_\"+str(x['pos']), axis=1)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "lo=LiftOver(\"hg38\", \"hg19\")\n",
    "\n",
    "def Generate_hg38_credset(study_ID, trait_name, output_path=\"./\", All_cred=Cred_SNPs):\n",
    "    Select_SNPs=Cred_SNPs.loc[Cred_SNPs['study_id'].str.contains(study_ID)].reset_index()\n",
    "    Select_SNPs=Select_SNPs[['tag_chrom', 'tag_pos']]\n",
    "    Select_SNPs['tag_chrom']=\"chr\"+Select_SNPs['tag_chrom']\n",
    "    Select_SNPs['hg38_chr']=Select_SNPs.apply(lambda x: Liftover_chr(input_chr=x['tag_chrom'], input_pos=x['tag_pos'], input_LO=lo), axis=1)\n",
    "    Select_SNPs['hg38_pos']=Select_SNPs.apply(lambda x: Liftover_pos(input_chr=x['tag_chrom'], input_pos=x['tag_pos'], input_LO=lo), axis=1)\n",
    "    Select_SNPs=Select_SNPs.loc[-Select_SNPs['hg38_chr'].str.contains(\"NA\")]\n",
    "\n",
    "    Select_SNPs['hg38_names']=Select_SNPs.apply(lambda x: x['hg38_chr']+\"_\"+str(x['hg38_pos']), axis=1)\n",
    "    Select_SNPs=Select_SNPs[['hg38_names', 'hg38_chr', 'hg38_pos']]\n",
    "\n",
    "    Select_SNPs.to_csv(output_path+study_ID+\"_\"+trait_name+\"_\"+\"hg19.txt\", sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "\n",
    "#Generate_hg38_credset(study_ID=\"FINNGEN_R5_L12_ATOPIC\", trait_name=\"\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"GCST002245\", trait_name=\"Alz\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"GCST004131\", trait_name=\"IBD\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"GCST005413\", trait_name=\"T2D\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"GCST005536\", trait_name=\"T1D\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"GCST005569\", trait_name=\"RA\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "\n",
    "#Generate_hg38_credset(study_ID=\"GCST001725\", trait_name=\"IBD\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"FINNGEN_R5_K11_IBD\", trait_name=\"IBD\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "#Generate_hg38_credset(study_ID=\"FINNGEN_R5_K11_IBD_STRICT\", trait_name=\"IBD\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "\n",
    "Generate_hg38_credset(study_ID=\"GCST005047\", trait_name=\"T2D\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "Generate_hg38_credset(study_ID=\"GCST006867\", trait_name=\"T2D\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "Generate_hg38_credset(study_ID=\"GCST007517\", trait_name=\"T2D\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "\n",
    "Generate_hg38_credset(study_ID=\"NEALE2_50_raw\", trait_name=\"Height\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n",
    "Generate_hg38_credset(study_ID=\"NEALE2_21001_raw\", trait_name=\"BMI\", output_path=\"/Users/xg1/Downloads/CHEERS/GWAS_inputs/\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": "'chr1'"
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Select_SNPs.iloc[1]['tag_chrom']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": "[('chr1', 119210549, '+', 20851231461)]"
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pyliftover import LiftOver\n",
    "lo=LiftOver(\"hg19\", \"hg38\")\n",
    "lo.convert_coordinate(Select_SNPs.iloc[3]['tag_chrom'], Select_SNPs.iloc[3]['tag_pos'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "lo=LiftOver(\"hg19\", \"hg38\")\n",
    "\n",
    "def Call_liftover(input_chr, input_pos, input_LO):\n",
    "    return(input_LO.convert_coordinate(input_chr, input_pos))\n",
    "\n",
    "def Liftover_chr(input_chr, input_pos, input_LO):\n",
    "    lifted_result=input_LO.convert_coordinate(input_chr, input_pos)\n",
    "    if(len(lifted_result) == 0):\n",
    "        return(\"NA\")\n",
    "    else:\n",
    "        return(lifted_result[0][0])\n",
    "\n",
    "def Liftover_pos(input_chr, input_pos, input_LO):\n",
    "    lifted_result=input_LO.convert_coordinate(input_chr, input_pos)\n",
    "    if(len(lifted_result) == 0):\n",
    "        return(\"NA\")\n",
    "    else:\n",
    "        return(lifted_result[0][1])\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import OrderedDict\n",
    "import json\n",
    "\n",
    "def variant_info_query(gene_id):\n",
    "  #construct query string and declare variables that will be sent in query\n",
    "  api_query = \"\"\"\n",
    "    query variantInfo($geneId: String!) {\n",
    "      variantInfo(variantId: $geneId) {\n",
    "        rsId\n",
    "      }\n",
    "    }\n",
    "  \"\"\" \n",
    "\n",
    "  #set base_url for Open Targets Genetics Portal API\n",
    "  base_url = \"https://api.genetics.opentargets.org/graphql\"\n",
    "\n",
    "  #set variables object\n",
    "  variables = {\"geneId\": gene_id}\n",
    "\n",
    "  #perform API call using query string and variables object \n",
    "  r = requests.post(base_url, json={\"query\": api_query, \"variables\": variables})\n",
    "\n",
    "  #check status code of GraphQL API response and print error message if code == 400\n",
    "  if str(r.status_code) == \"400\":\n",
    "    print(f\"{gene_id} query status code: {r.status_code}\")\n",
    "  else:\n",
    "    pass\n",
    "\n",
    "  #transform API response into JSON \n",
    "  api_response_as_json = json.loads(r.text)\n",
    "  \n",
    "  #print first element of JSON response data\n",
    " # return(api_response_as_json[\"data\"][\"variantInfo\"]['rsId'])\n",
    "  if(len(api_response_as_json) == 1):\n",
    "    return(api_response_as_json[\"data\"][\"variantInfo\"]['rsId'])\n",
    "  else:\n",
    "    return(\"NA\")\n",
    "#variant_info_query(\"1_1182895_C_T\")\n",
    "\n",
    "def make_query_inputs(chrom, pos, ref, alt):\n",
    "    input_var=str(chrom)+\"_\"+str(pos)+\"_\"+str(ref)+\"_\"+str(alt)\n",
    "    return(variant_info_query(input_var))\n",
    "\n",
    "def Make_LDSC_sumstats(input_sumstats, out_dir, out_prefix):\n",
    "    raw_sumstats=pd.read_parquet(input_sumstats)\n",
    "    raw_sumstats['SNP']=raw_sumstats.apply(lambda x: make_query_inputs(x['chrom'], x['pos'], x['ref'], x['alt']), axis=1)\n",
    "    cols = OrderedDict([\n",
    "        ('SNP', 'snpid'),\n",
    "        ('ref', 'a1'),\n",
    "        ('alt', 'a2'),\n",
    "        ('n_total', 'N'),\n",
    "        ('pval', 'P'),\n",
    "        ('eaf', 'Freq.Allele1')\n",
    "        ])\n",
    "    out_sumstats=raw_sumstats.loc[:, list(cols.keys())].rename(columns=cols)\n",
    "    out_sumstats.to_csv(out_dir+out_prefix+\"_sumstats.txt\", sep=\"\\t\", index=False)\n",
    "\n",
    "    \n",
    "\n",
    "#IBD_sumstats['SNP']=IBD_sumstats.apply(lambda x: make_query_inputs(x['chrom'], x['pos'], x['ref'], x['alt']), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_sumstats=pd.read_parquet(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST002245.parquet\")\n",
    "print(raw_sumstats)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import OrderedDict\n",
    "\n",
    "def Make_LDSC_sumstats(input_sumstats, out_dir, out_prefix):\n",
    "    raw_sumstats=pd.read_parquet(input_sumstats)\n",
    "    cols = OrderedDict([\n",
    "        ('rs_id', 'snpid'),\n",
    "        ('ref', 'a1'),\n",
    "        ('alt', 'a2'),\n",
    "        ('n_total', 'N'),\n",
    "        ('pval', 'P'),\n",
    "        ('eaf', 'Freq.Allele1')\n",
    "        ])\n",
    "    out_sumstats=raw_sumstats.loc[:, list(cols.keys())].rename(columns=cols)\n",
    "    out_sumstats.to_csv(out_dir+out_prefix+\"_sumstats.txt\", sep=\"\\t\", index=False)\n",
    "\n",
    "# Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST002245.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"Alzheimers\")\n",
    "# Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/ATOPIC.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"Atopic\")\n",
    "\n",
    "# Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST005413.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T2D\")\n",
    "# Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST005536.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T1D\")\n",
    "# #Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/GCST005569.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"RA\")\n",
    "# Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST004131.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"IBD\")\n",
    "\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/xg1_rsid_sumstats_FINNGEN_R5_K11_IBD_STRICT.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"IBD_16\")\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/FINNGEN_R5_K11_IBD.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"IBD_10\")\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST001725.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"IBD_76\")\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST005047.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T2D_29\")\n",
    "Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST006867.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T2D_131\")\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/GCST007517.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T2D_46\")\n",
    "\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/NEALE2_50_raw.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"Height\")\n",
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/with_rsid/NEALE_21001_raw.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"BMI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "start_time=time.time()\n",
    "Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/GCST002245.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"Alzheimers\")\n",
    "Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/FINNGEN_R5_L12_ATOPIC.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"Atopic\")\n",
    "\n",
    "Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/GCST005413.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T2D\")\n",
    "Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/GCST005536.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"T1D\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/GCST005569.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"RA\")\n",
    "Make_LDSC_sumstats(\"/Users/xg1/Downloads/ldsc/input_sumstats/GCST004131.parquet\", \"/Users/xg1/Downloads/ldsc/Formatted_sumstats/\", \"IBD\")\n",
    "print(time.time()-start_time)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Sumstat_annot_dir=\"/Users/xg1/Downloads/ldsc/Comparison_baseline/\"\n",
    "#print(os.listdir(Sumstat_annot_dir))\n",
    "Test_samples=[\"BSS00121\", \"BSS00122\", \"BSS00123\", \"BSS00206\", \"BSS00207\", \"BSS01712\", \"BSS01714\", \"BSS00196\", \"BSS00197\", \"BSS00188\", \"BSS00189\", \"BSS00233\", \"BSS00234\", \"BSS00227\", \"BSS00228\", \"BSS01113\", \"BSS01120\", \"BSS00084\", \"BSS00330\", \"BSS00079\", \"BSS00080\", \"BSS01665\", \"BSS01666\", \"BSS01169\", \"BSS00511\", \"BSS01068\", \"BSS01071\"]\n",
    "for i in range(1, 23):\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Input_SNPs=pd.read_csv(\"/home/xg1/CHEERS/input_SNPs/Filtered_GCST004131_hg38.nopp.input.bed\",\n",
    "                      sep='\\t', names=[\"name\", \"chr\", \"pos\"])\n",
    "Input_SNPs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Input_SNPs['name']=Input_SNPs['name']+\"_\"+str(Input_SNPs['pos'])\n",
    "abc=Input_SNPs.apply(lambda x: x['name']+\"_\"+str(x['pos']), axis=1)\n",
    "#Input_SNPs.apply(lambda x: print(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "Input_SNPs['name']=abc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "Input_SNPs.to_csv(\"/home/xg1/CHEERS/input_SNPs/Correct_IBD_SNPs.txt\",\n",
    "                 sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3218820, 11)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Cred_SNPs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "## PICS output is in ld.parquet\n",
    "PICS_SNPs=pd.read_parquet(\"/home/xg1/genetics-v2d-data/output/210601/ld.parquet/\")\n",
    "\n",
    "## sumstat credsets in finemapping.parquet:\n",
    "Cred_SNPs=pd.read_parquet(\"/Users/xg1/Downloads/finemapping.parquet\",\n",
    "                engine='pyarrow')\n",
    "\n",
    "## load up study table and search for studies which match a EFO\n",
    "study_table=pd.read_parquet(\"/home/xg1/genetics-v2d-data/output/210601/studies.parquet\")\n",
    "\n",
    "Trait_matched_studies=study_table.loc[study_table.trait_efos.apply(lambda x: \"EFO_0000685\" in x)]\n",
    "\n",
    "Trait_matched_sumstats=Trait_matched_studies.loc[Trait_matched_studies['has_sumstats'] == True]['study_id'].unique()\n",
    "\n",
    "\n",
    "Trait_matched_PICS=Trait_matched_studies.loc[Trait_matched_studies['has_sumstats'] == False]['study_id'].unique()\n",
    "\n",
    "\n",
    "## For studies which have sumstats, look up credible SNPs in finemapping.parquet\n",
    "\n",
    "Trait_sumstat_SNPs=Cred_SNPs.loc[Cred_SNPs['study_id'].isin(Trait_matched_sumstats)]\n",
    "Trait_sumstat_SNPs.drop(['log10_ABF'], axis=1)\n",
    "\n",
    "## Otherwise get credible SNPs from ld.parquet\n",
    "\n",
    "Trait_PICS_SNPs=PICS_SNPs.loc[PICS_SNPs['study_id'].isin(Trait_matched_PICS)]\n",
    "Trait_PICS_SNPs=Trait_PICS_SNPs.drop(['overall_r2', 'pics_mu', 'pics_95perc_credset', 'pics_99perc_credset', 'AFR_1000G_prop',\n",
    "                      'AMR_1000G_prop', 'EAS_1000G_prop', 'EUR_1000G_prop', 'SAS_1000G_prop' ], axis=1)\n",
    "Trait_PICS_SNPs=Trait_PICS_SNPs.rename(columns={'pics_postprob':'posterior_prob'})\n",
    "## Merge the SNPs retrieved together, keep unique credible SNPs\n",
    "\n",
    "Trait_all_SNPs=pd.concat([Trait_sumstat_SNPs, Trait_PICS_SNPs])\n",
    "\n",
    "## Assign names to each row (Chr+pos as a string is fine)\n",
    "\n",
    "Trait_all_SNPs_pp=Trait_all_SNPs[['tag_chrom', 'tag_pos', 'posterior_prob']]\n",
    "\n",
    "Traits_all_SNPs_pp['name']=Traits_all_SNPs_pp.apply(lambda x: x['name']+\"_\"+str(x['pos']), axis=1)\n",
    "\n",
    "\n",
    "## Output\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Trait_all_SNPs_pp=Trait_all_SNPs[['tag_chrom', 'tag_pos', 'posterior_prob']]\n",
    "\n",
    "Traits_all_SNPs_pp['name']=Traits_all_SNPs_pp.apply(lambda x: x['tag_chrom']+\"_\"+str(x['tag_pos']), axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_id</th>\n",
       "      <th>lead_chrom</th>\n",
       "      <th>lead_pos</th>\n",
       "      <th>lead_ref</th>\n",
       "      <th>lead_alt</th>\n",
       "      <th>tag_chrom</th>\n",
       "      <th>tag_pos</th>\n",
       "      <th>tag_ref</th>\n",
       "      <th>tag_alt</th>\n",
       "      <th>posterior_prob</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31213</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>1</td>\n",
       "      <td>17347613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>17347613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.214298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31214</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>1</td>\n",
       "      <td>17347613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>17346235</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.213230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31215</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>1</td>\n",
       "      <td>17347613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>17348042</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.145258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31216</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>1</td>\n",
       "      <td>17347613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>17347690</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.132471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31217</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>1</td>\n",
       "      <td>17347613</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>17330628</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.096870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3063173</th>\n",
       "      <td>FINNGEN_R5_M13_RHEUMA</td>\n",
       "      <td>8</td>\n",
       "      <td>128503195</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>8</td>\n",
       "      <td>128703972</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3063174</th>\n",
       "      <td>FINNGEN_R5_M13_RHEUMA</td>\n",
       "      <td>8</td>\n",
       "      <td>128503195</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>8</td>\n",
       "      <td>128709557</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3110695</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>9</td>\n",
       "      <td>34710263</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>9</td>\n",
       "      <td>34710263</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.461193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3110696</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>9</td>\n",
       "      <td>34710263</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>9</td>\n",
       "      <td>34710341</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.263071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3110697</th>\n",
       "      <td>GCST005569</td>\n",
       "      <td>9</td>\n",
       "      <td>34710263</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>9</td>\n",
       "      <td>34707376</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.259484</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1460 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      study_id lead_chrom   lead_pos lead_ref lead_alt  \\\n",
       "31213               GCST005569          1   17347613        C        T   \n",
       "31214               GCST005569          1   17347613        C        T   \n",
       "31215               GCST005569          1   17347613        C        T   \n",
       "31216               GCST005569          1   17347613        C        T   \n",
       "31217               GCST005569          1   17347613        C        T   \n",
       "...                        ...        ...        ...      ...      ...   \n",
       "3063173  FINNGEN_R5_M13_RHEUMA          8  128503195        T        C   \n",
       "3063174  FINNGEN_R5_M13_RHEUMA          8  128503195        T        C   \n",
       "3110695             GCST005569          9   34710263        G        A   \n",
       "3110696             GCST005569          9   34710263        G        A   \n",
       "3110697             GCST005569          9   34710263        G        A   \n",
       "\n",
       "        tag_chrom    tag_pos tag_ref tag_alt  posterior_prob  \n",
       "31213           1   17347613       C       T        0.214298  \n",
       "31214           1   17346235       G       A        0.213230  \n",
       "31215           1   17348042       C       A        0.145258  \n",
       "31216           1   17347690       C       T        0.132471  \n",
       "31217           1   17330628       A       G        0.096870  \n",
       "...           ...        ...     ...     ...             ...  \n",
       "3063173         8  128703972       T       C        0.001159  \n",
       "3063174         8  128709557       A       C        0.001077  \n",
       "3110695         9   34710263       G       A        0.461193  \n",
       "3110696         9   34710341       G       A        0.263071  \n",
       "3110697         9   34707376       G       A        0.259484  \n",
       "\n",
       "[1460 rows x 10 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Trait_sumstat_SNPs.drop(['log10_ABF'], axis=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "Trait_all_SNPs=pd.concat([Trait_sumstat_SNPs.drop(['log10_ABF'], axis=1)\n",
    ", Trait_PICS_SNPs.drop(['overall_r2', 'pics_mu', 'pics_95perc_credset', 'pics_99perc_credset', 'AFR_1000G_prop',\n",
    "                      'AMR_1000G_prop', 'EAS_1000G_prop', 'EUR_1000G_prop', 'SAS_1000G_prop' ], axis=1)])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "def example_query(studyId, variantId):\n",
    "  #construct query string and declare variables that will be sent in query\n",
    "  api_query = \"\"\"\n",
    "    query StudyLocusQTLColocQuery($studyId: String!, $variantId: String!) {\n",
    "    variantInfo(variantId: $variantId) {\n",
    "        id\n",
    "        rsId\n",
    "        chromosome\n",
    "        position\n",
    "        refAllele\n",
    "        altAllele\n",
    "    }\n",
    "    qtlColocalisation(studyId: $studyId, variantId: $variantId) {\n",
    "        indexVariant {\n",
    "        id\n",
    "        rsId\n",
    "        chromosome\n",
    "        position\n",
    "        refAllele\n",
    "        altAllele\n",
    "        }\n",
    "        gene {\n",
    "        id\n",
    "        symbol\n",
    "        chromosome\n",
    "        start\n",
    "        end\n",
    "        bioType\n",
    "        }\n",
    "        phenotypeId\n",
    "        tissue {\n",
    "        id\n",
    "        name\n",
    "        }\n",
    "        qtlStudyName\n",
    "        beta\n",
    "        h3\n",
    "        h4\n",
    "        log2h4h3\n",
    "    }\n",
    "    }\n",
    "  \"\"\" \n",
    "  #set base_url for Open Targets Genetics Portal API\n",
    "  base_url = \"https://api.genetics.opentargets.org/graphql\"\n",
    "  #set variables object\n",
    "  variables = {\"studyId\": studyId, \"variantId\": variantId}\n",
    "  #perform API call using query string and variables object \n",
    "  r = requests.post(base_url, json={\"query\": api_query, \"variables\": variables})\n",
    "  if str(r.status_code) == \"400\":\n",
    "    print(f\"query status code: {r.status_code}\")\n",
    "  else:\n",
    "    pass\n",
    "  api_response_as_json = json.loads(r.text)\n",
    "  return(api_response_as_json[\"data\"])\n",
    "All_IL18RAP_coloc_QTLs=example_query(\"FINNGEN_R5_ATOPIC_STRICT_REIMB\",  \"2_102476784_T_C\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "All_IL18RAP_coloc_QTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2.7.16 64-bit",
   "name": "python2716jvsc74a57bd0545e036c4b32438aced1f6b3c8d38ca151d9c36189e05839cb0aa568fda70ddd"
  },
  "language_info": {
   "name": "python",
   "version": ""
  },
  "metadata": {
   "interpreter": {
    "hash": "545e036c4b32438aced1f6b3c8d38ca151d9c36189e05839cb0aa568fda70ddd"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}